debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SourceInfo.java

```diff
@@ -127,6 +128,8 @@
                                                      .field(THREAD_KEY, Schema.OPTIONAL_INT64_SCHEMA)
                                                      .field(DB_NAME_KEY, Schema.OPTIONAL_STRING_SCHEMA)
                                                      .field(TABLE_NAME_KEY, Schema.OPTIONAL_STRING_SCHEMA)
+                                                     // TODO how to make this optional?
```

Comment:
Since this is built statically, is it possible to alter the schema instance to drop the field when it has no value?  I thought we could modify it in the schema() method below, but that seems like building in a fairly nasty side effect based on currentQuery being null or not.

Reply:
The schema would always contain "query" as an optional String field. But the code for creating the struct would only populate that field if there's a non-null value.

Reply:
So this would seem to be doing that correctly then? https://github.com/debezium/debezium/pull/523/files#diff-3a15e1a239968077827d2be8142eb3cdR348

Reply:
Yes, that code looks alright. Has this changed since yesterday? Because I definitely saw `"query ": null` in messages when building the connector with that PR and running it in Kafka Connect.

Reply:
Same code as yesterday, and no you aren't crazy :) . The tests I added confirm that `"query": null` is present in the messages if that field is not set.

Reply:
Let me re-run the test cases w/ a breakpoint and just verify for myself that the statement is evaluating as false

Reply:
Ok, that's re-assuring that I didn't dream it :)

Reply:
Hmm maybe **I'm** the crazy one, running through the test cases that should not include the query property, it's not showing up in the SourceRecord.toString() output.  Perhaps I didn't have the if statement yesterday after all?

I'm about to head out for the day, but I can spend some additional time tomorrow and validate this is working as expected across more test cases.

Reply:
OK took another look at this.  When the query field is not populated the resulting struct looks like:

![image](https://user-images.githubusercontent.com/571653/40759272-5fe87ad6-64cc-11e8-8b22-190f722fb1ad.png)

Compared to when it IS populated with a value:

![image](https://user-images.githubusercontent.com/571653/40759307-7cbfcfa6-64cc-11e8-8d33-69b5b94c3e17.png)

I believe this matches with the behavior you were looking for correct?

Reply:
Yeah, all looks good now. I'm going to remove the TODO comment before merging.